internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -150,11 +230,114 @@ func (c *controller) extractValueAsJWTAuthenticator(value authncache.Value) *jwt
 	return jwtAuthenticator
 }
 
-// newJWTAuthenticator creates a jwt authenticator from the provided spec.
-func newJWTAuthenticator(spec *auth1alpha1.JWTAuthenticatorSpec) (*jwtAuthenticator, error) {
-	rootCAs, _, err := pinnipedauthenticator.CABundle(spec.TLS)
-	if err != nil {
-		return nil, fmt.Errorf("invalid TLS configuration: %w", err)
+func (c *jwtCacheFillerController) updateStatus(
+	ctx context.Context,
+	original *auth1alpha1.JWTAuthenticator,
+	conditions []*metav1.Condition,
+) error {
+	updated := original.DeepCopy()
+
+	if conditionsutil.HadErrorCondition(conditions) {
+		updated.Status.Phase = auth1alpha1.JWTAuthenticatorPhaseError
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonNotReady,
+			Message: "the JWTAuthenticator is not ready: see other conditions for details",
+		})
+	} else {
+		updated.Status.Phase = auth1alpha1.JWTAuthenticatorPhaseReady
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeReady,
+			Status:  metav1.ConditionTrue,
+			Reason:  reasonSuccess,
+			Message: "the JWTAuthenticator is ready",
+		})
+	}
+
+	_ = conditionsutil.MergeConfigConditions(
+		conditions,
+		original.Generation,
+		&updated.Status.Conditions,
+		plog.New().WithName(controllerName),
+		metav1.NewTime(c.clock.Now()),
+	)
+
+	if equality.Semantic.DeepEqual(original, updated) {
+		return nil
+	}
+	_, err := c.client.AuthenticationV1alpha1().JWTAuthenticators().UpdateStatus(ctx, updated, metav1.UpdateOptions{})
+	return err
+}
+
+func (c *jwtCacheFillerController) validateJWKSFetch(ctx context.Context, jwksURL string, conditions []*metav1.Condition, prereqOk bool) (*coreosoidc.RemoteKeySet, []*metav1.Condition, error) {
+	if !prereqOk {
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSFetchValid,
+			Status:  metav1.ConditionUnknown,
+			Reason:  reasonUnableToValidate,
+			Message: msgUnableToValidate,
+		})
+		return nil, conditions, nil
+	}
+	keySet := coreosoidc.NewRemoteKeySet(ctx, jwksURL)
+	testJWTToken := jwt.NewWithClaims(jwt.SigningMethodNone, jwt.MapClaims{
+		"aud": "fake-audience-for-verification-probe",
+	})
+	rawToken, signErr := testJWTToken.SignedString(jwt.UnsafeAllowNoneSignatureType)
+	// no unit tests for this block.
+	// this is not configurable, there is no way to change the token we are using
+	// for testing, so we simply shouldn't hit this block.
+	if signErr != nil {
+		errText := "could not sign tokens"
+		msg := fmt.Sprintf("%s: %s", errText, signErr.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSFetchValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidTokenSigningFailure,
+			Message: msg,
+		})
+		return keySet, conditions, fmt.Errorf("%s: %w", errText, signErr)
+	}
+	_, verifyWithKeySetErr := keySet.VerifySignature(ctx, rawToken)
+	verifyErrString := verifyWithKeySetErr.Error()
```

Comment:
What if `verifyWithKeySetErr` was nil? Could this line crash? Should we handle nil differently?

Reply:
I think since this JWT will never be correctly verified with the server keys, since we made it up, there should always be some kind of error?  I can definitely add a defensive block just in case.